### PR TITLE
Add RenderEditable keys listener back in attach()

### PR DESCRIPTION
**Pull request by tgucio**
## Description
RenderEditable currently shows an issue when it's detached and re-attached to the root tree, resulting in navigation and shortcut keys stopping to work (while text input still works).
This appears to be because the detach() function calls RawKeyboard.instance.removeListener(_handleKeyEvent) and on re-attaching, RawKeyboard.instance.addListener(_handleKeyEvent) isn't called - and as a result even though setFocus() is called, the key event handler is missing forever.
This PR adds the key event listener in attach() if _listenerAttached is true, meaning the widget was previously attached and handling keys.

## Related Issues
https://github.com/flutter/flutter/issues/75562

## Tests
I added the following tests:
```
packages/flutter/test/rendering/editable_test.dart:
test('arrow keys and delete handle surrogate pairs correctly'
```

Relevant tests passing:
```
% ../../bin/flutter test test/rendering/editable_test.dart
00:03 +53: All tests passed!                                                                                                                
% ../../bin/flutter test test/rendering/editable_gesture_test.dart 
00:03 +1: All tests passed!                                                                                                                 
% ../../bin/flutter test test/widgets/editable_text_test.dart 
00:12 +161: All tests passed!                                                                                                               
% ../../bin/flutter test test/widgets/editable_text_cursor_test.dart 
00:07 +24: All tests passed!                                                                                                                
% ../../bin/flutter test test/widgets/editable_text_show_on_screen_test.dart 
00:04 +9: All tests passed!                                                                                                                 
```

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/flutter/test/rendering/editable_test.dart [hunk 1]
```diff
@@ -955,6 +955,67 @@ void main() {
...
+        offset: 0,
+      ),
+    );
+
+    final PipelineOwner pipelineOwner = PipelineOwner();
+    editable.attach(pipelineOwner);
+    editable.hasFocus = true;
+    editable.detach();
+    layout(editable);
+    editable.hasFocus = true;
```
Comment (LongCatIsLooong) [inline packages/flutter/test/rendering/editable_test.dart:993]: nit: are both `hasFocus = true` needed?

File: packages/flutter/test/rendering/editable_test.dart [hunk 2]
```diff
@@ -955,6 +955,67 @@ void main() {
...
+    editable.selectPositionAt(from: Offset.zero, cause: SelectionChangedCause.tap);
```
Comment (LongCatIsLooong) [inline packages/flutter/test/rendering/editable_test.dart:994]: nit: are line 994 and line 995 essentially doing the same thing (setting the selection to `(0, 0)`)?

**Review by LongCatIsLooong — APPROVED** [review]
LGTM. Kinda hard to believe we had this issue all this time. Thank you for fixing this!

File: packages/flutter/test/rendering/editable_test.dart (hunk 1 above)
Comment (tgucio) [reply packages/flutter/test/rendering/editable_test.dart:993]: The hasFocus setter has logic to add the keyboard listener the first time the widget is focused. This was to show the test was failing depite setFocus being called after attaching again. And also typically after reattaching the widget would be focused again by the user (e.g. by clicking on it).

File: packages/flutter/test/rendering/editable_test.dart (hunk 2 above)
Comment (tgucio) [reply packages/flutter/test/rendering/editable_test.dart:994]: This was pretty much copied from other tests.
